apps/tx-builder/src/components/utils/Confirmable.tsx

```diff
@@ -0,0 +1,78 @@
+import ReactDOM from 'react-dom'
+import React from 'react'
+import { ThemeProvider } from 'styled-components'
+import { theme } from '@gnosis.pm/safe-react-components'
+
+type ConfirmationProps = {
+  proceed: (value: boolean) => void
+  show: boolean
+}
+
+type Confirmable = {
+  resolve: (value: boolean) => void
+}
+
+const confirmable =
+  <ComponentProps extends ConfirmationProps>(
+    Component: React.FC<ComponentProps>,
+  ): React.FC<Confirmable & Omit<ComponentProps, keyof ConfirmationProps>> =>
+  props => {
+    const [show, setShow] = React.useState(true)
+    const { resolve, ...rest } = props
+
+    const proceed = (value: boolean) => {
+      setShow(false)
+      resolve(value)
+    }
+
+    if (show) {
+      // @ts-ignore
+      return <Component proceed={proceed} show={show} {...rest} />
+    } else {
+      return null
+    }
+  }
+
+const createConfirmation =
+  <ComponentProps extends Confirmable>(
+    Component: React.FC<ComponentProps>,
+    unmountDelay = 1000,
+    mountingNode = document.body,
+  ) =>
+  (props: Omit<ComponentProps, keyof Confirmable>) => {
+    const wrapper = mountingNode.appendChild(document.createElement('div'))
+
+    const promise = new Promise<boolean>(resolve => {
+      try {
+        ReactDOM.render(
+          <ThemeProvider theme={theme}>
+            {/* @ts-ignore */}
```

Comment:
would be great if someone could check if it's possible to get rid of `ts-ignore`, couldn't figure it out